src/shared/test/frame-rpc-test.js

```diff
@@ -0,0 +1,146 @@
+import { RPC } from '../frame-rpc';
+
+describe('shared/bridge', () => {
+  let port1;
+  let port2;
+  let rpc1;
+  let rpc2;
+  let plusOne;
+
+  beforeEach(() => {
+    const channel = new MessageChannel();
+    port1 = channel.port1;
+    port2 = channel.port2;
+
+    // `concat` method for rpc1
+    const concat = (arg0, ...args) => {
+      const callback = args.pop();
+      const result = arg0.concat(...args);
+      callback(result);
+    };
+
+    rpc1 = new RPC(
+      /** dummy when using ports */ window,
+      port1,
+      /** dummy when using ports */ '*',
+      {
+        concat,
+      }
+    );
+
+    // `plusOne` method for rpc2
+    plusOne = sinon.stub().callsFake((...numbers) => {
+      const callback = numbers.pop();
+      const result = numbers.map(value => value + 1);
+      callback(result);
+    });
+
+    rpc2 = new RPC(
+      /** dummy when using ports */ window,
+      port2,
+      /** dummy when using ports */ '*',
+      {
+        plusOne,
+      }
+    );
+  });
+
+  afterEach(() => {
+    rpc1.destroy();
+    rpc2.destroy();
+  });
+
+  it('should call the method `plusOne` on rpc2', done => {
+    rpc1.call('plusOne', 1, 2, 3, value => {
+      assert.deepEqual(value, [2, 3, 4]);
+      done();
+    });
+  });
+
+  it('should not call the method `plusOne` if rpc1 is destroyed', () => {
+    rpc1.destroy();
+
+    rpc1.call('plusOne', 1, 2, 3);
+    assert.notCalled(plusOne);
+  });
+
+  it('should not call the method `plusOne` if rpc2 is destroyed', done => {
+    rpc2.destroy();
+
+    rpc1.call('plusOne', 1, 2, 3, () => {
+      done(new Error('Unexpected call'));
+    });
+
+    setTimeout(() => {
+      assert.notCalled(plusOne);
+      done();
+    }, 100);
```

Comment:
You don't need a 100ms timeout here. 0 is fine. `postMessage` enqueues a task to be run on a subsequent event loop turn, so all you need to do is use `setTimeout(callback, 0)` to enqueue a final task on the queue which will run after all the `postMessage` tasks have been delivered.